lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -1017,7 +1020,7 @@ def cluster_password_synced(self) -> bool:
 
         try:
             # check our ability to use connect to both mongos and our current replica set.
-            mongos_reachable = self._is_mongos_reachable()
+            mongos_reachable = self._is_mongos_reachable(with_auth=True)
             with MongoDBConnection(self.charm.mongodb_config) as mongo:
                 mongod_reachable = mongo.is_ready
         except OperationFailure as e:
```

Comment:
Is there are reason why we `raise` 3 lines below this one? as opposed to returning `False` (seeing that this method is only called in `update_status`)

Similary on `ShardingProvider.cluster_password_synced`

Reply:
Yes - errors outside of: Authentication and Failures with Internal membership should be reported + raised - because they are related to something different entirely.

This is used in update status only to report Authentication Failures with Failures Internal membership